packages/babel-generator/src/generators/typescript.ts

```diff
@@ -51,6 +51,11 @@ export function TSTypeParameterInstantiation(
 export { TSTypeParameterInstantiation as TSTypeParameterDeclaration };
 
 export function TSTypeParameter(this: Printer, node: t.TSTypeParameter) {
+  if (node.const) {
+    this.word("const");
+    this.space();
+  }
+
```

Comment:
The `tsc` does not enforce ordering between `const` and `in`/`out`, in other words both `<const in T>` and `<in const T>` are valid in typescript. Here we print `const` before `in` so that modifiers will appear in alphabetical ordering.